uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/RouteContextSupportTest.java

```diff
@@ -0,0 +1,156 @@
+package org.jboss.tools.fuse.ui.bot.tests;
+
+import static org.jboss.tools.fuse.reddeer.wizard.NewFuseIntegrationProjectWizardDeploymentType.STANDALONE;
+import static org.jboss.tools.fuse.reddeer.wizard.NewFuseIntegrationProjectWizardRuntimeType.KARAF;
+import static org.junit.Assert.fail;
+import java.util.Collection;
+import org.eclipse.reddeer.common.exception.WaitTimeoutExpiredException;
+import org.eclipse.reddeer.common.wait.AbstractWait;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitUntil;
+import org.eclipse.reddeer.eclipse.condition.ConsoleHasText;
+import org.eclipse.reddeer.eclipse.ui.views.contentoutline.ContentOutline;
+import org.eclipse.reddeer.junit.runner.RedDeerSuite;
+import org.jboss.tools.fuse.reddeer.ProjectTemplate;
+import org.jboss.tools.fuse.reddeer.editor.CamelEditor;
+import org.jboss.tools.fuse.reddeer.editor.SourceEditor;
+import org.jboss.tools.fuse.reddeer.projectexplorer.CamelProject;
+import org.jboss.tools.fuse.reddeer.utils.LogChecker;
+import org.jboss.tools.fuse.reddeer.utils.ProjectFactory;
+import org.jboss.tools.fuse.reddeer.wizard.CamelXmlFileWizard;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.eclipse.reddeer.swt.api.Shell;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.impl.button.FinishButton;
+import org.eclipse.reddeer.workbench.impl.shell.WorkbenchShell;
+
+/**
+ * @author fpospisI
+ */
+@RunWith(RedDeerSuite.class)
+public class RouteContextSupportTest {
+	public static final String PROJECT_NAME = "routeContextSupportTest";
+	public static final String ROUTES_NAME = "camel-routes.xml";
+	
+	/**
+	 * <p>
+	 * routeContext support test
+	 * </p>
+	 * <b>Steps</b>
+	 * <ol>
+	 * <li>Create new project: STANDALONE - KARAF - CBR_SPRING.</li>
+	 * <li>In source tab change XML to external <routeContext> "camel-routes.xml" reference.</li>
+	 * <li>Create new XML - camel-routes.xml.</li>
+	 * <li>Edit 'Source' in camel-routes.xml.</li>
+	 * <li>Check camel-routes.xml 'Design' tab for containing items setted up in source. </li>
+	 * <li>Run as Local Camel Context (without tests) and check for Fuse related errors.</li>
+	 * </ol>
+	 * 
+	 * @author fpospisi
+	 */
+	@Test
+	public void testRouteContextSupport() {
+		/*
+		 * Create new project.
+		 * */
+		ProjectFactory.newProject(PROJECT_NAME).deploymentType(STANDALONE).runtimeType(KARAF).template(ProjectTemplate.CBR_SPRING).create();
+		
+		/*
+		 * Change content in 'Source'tab for main file.
+		 */
+		CamelEditor.switchTab("Source");
+		AbstractWait.sleep(TimePeriod.getCustom(3));
+		SourceEditor editor = new SourceEditor();
+		editor.selectText(0, 999999);		
+		editor.insertText("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
```

Comment:
I would prefer to have XML code in file inside `resources` folder and then use method `EditorManipulator.copyFileContentToCamelXMLEditor` to handle the same thing. WDYT?

It is beneficial because 
- code will be cleaner and more readable
- if you will want to test manually these test workflow you won't have to rewrite context.xml files from code but only reuse existing files from `resources` folder